core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -208,7 +209,9 @@ class BrokerToControllerRequestThread(networkClient: KafkaClient,
   }
 
   private[server] def handleResponse(request: BrokerToControllerQueueItem)(response: ClientResponse): Unit = {
-    if (response.wasDisconnected()) {
+    if (isTimedOut(response)) {
+      debug(s"The request failed to send to the controller after timeout ${clientRequestTimeout} ms: $response")
+    } else if (response.wasDisconnected()) {
```

Comment:
(The following question is not related to this PR. just curios)

Is it possible to encounter authentication error when forwarding? If so, is it necessary to avoid retrying such (error) request?

Reply:
Thanks, for authorization related exception, we could add the logic to stop retry as well.

Reply:
Note that we handle authentication exceptions in `InterBrokerSendThread`. We log a message and mark the response as "disconnected." I think it is reasonable to keep retrying in case the broker's configuration changes and so that the error messages will remain persistent in the logs.